### PR TITLE
[NETBEANS-3786] : Fixed position of hint (add import) for record (jdk14). 

### PR DESCRIPTION
**Pull request by singh-akhilesh**
https://issues.apache.org/jira/browse/NETBEANS-3786

This issue occurs only when there is no any other import statement present

### CODE REVIEW

File: java/java.source.base/src/org/netbeans/modules/java/source/save/PositionEstimator.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
      * For JCTree instances that have no associated source position.
      */
     public static final int NOPOS = -2;
-    
+    public static final String RECORD = "RECORD";
```
[the remark this answers is not in the document]
Comment (singh-akhilesh) [reply java/java.source.base/src/org/netbeans/modules/java/source/save/PositionEstimator.java]: comment addressed

File: java/java.source.base/src/org/netbeans/modules/java/source/save/PositionEstimator.java [hunk 2]
```diff
@@ -314,7 +314,7 @@ public int prepare(final int startPos, StringBuilder aHead, StringBuilder aTail)
                 do {
                     Tree t = cut.getTypeDecls().get(tdpos);
                     typeDeclStart = (int) positions.getStartPosition(cut, t);
-                    if (TreeUtilities.CLASS_TREE_KINDS.contains(t.getKind())) {
+                    if (TreeUtilities.CLASS_TREE_KINDS.contains(t.getKind()) || t.getKind().toString().equals(RECORD)) {
```
[the remark this answers is not in the document]
Comment (singh-akhilesh) [reply java/java.source.base/src/org/netbeans/modules/java/source/save/PositionEstimator.java]: comment addressed

**Comment by singh-akhilesh** [comment]
@ebarboni Hi Erik, could you accommodate this PR.

**Comment by ebarboni** [comment]
hi @singh-akhilesh, would be perfect to rebase on your side. Will merge after

**Comment by singh-akhilesh** [comment]
> hi @singh-akhilesh, would be perfect to rebase on your side. Will merge after

Rebase done

**Pull request merged**